### PR TITLE
[pkg/stanza] Initialize with log-collection as a subtree

### PR DESCRIPTION
**Pull request by djaglowski**
This also temporarily excludes `pkg/stanza` from several validation checks. Checklist for reverting those changes:
- [ ] Remove exclusion from `Makefile`'s `ALL_MODULES`
- [ ] Remove exclusion from `.github/workflows/scripts/check-collector-module-version.sh`'s `mod_files`
- [ ] Remove exclusion from `versions.yaml`

### CODE REVIEW

File: Makefile.Common
Comment (djaglowski) [inline Makefile.Common:120]: This change also implemented on #9946

File: pkg/stanza/.github/codecov.yaml
Comment (mx-psi) [inline pkg/stanza/.github/codecov.yaml:1]: Will this and other .github configuration files be removed in a follow-up PR?

**Review by mx-psi — APPROVED** [review]
I didn't participate on the previous "move git history across repos" efforts, but this looks sensible to me. I think we should create an issue for follow-up steps. Also, I left a comment about when do we want to remove/move files related to Github Actions pipelines for log-collection

File: pkg/stanza/.github/codecov.yaml
Comment (djaglowski) [reply pkg/stanza/.github/codecov.yaml:1]: Yes, definitely. My next PR will be to do exactly that.

**Comment by djaglowski** [comment]
#9946 should be decided on before this PR is merged, since this PR includes that change.

**Review by pmm-sumo — APPROVED** [review]
Impressive collection of commits

File: pkg/stanza/.github/codecov.yaml
Comment (djaglowski) [reply pkg/stanza/.github/codecov.yaml:1]: I've created #9960 as an aggregate tracking issue.

**Comment by bogdandrutu** [comment]
@djaglowski you need to manually cleanup all the commits (the last once) since we want to do a "merge" not squash commit and all commits will be kept forever. I think you can at least start by dropping the impi merge, and ping on this issue when list is clean.

**Comment by djaglowski** [comment]
@bogdandrutu, this should be good to go now, if you want to have another look.

**Pull request merged**
